### PR TITLE
Silence dead code warnings in consensus

### PR DESCRIPTION
**Pull request by banool**
### Description
These warnings have been here for a long time. I'll open an issue to clean these up later.

### Test Plan
```
cargo run -p aptos-node
```

### CODE REVIEW

**Review by sherry-x — APPROVED** [review]

File: consensus/src/state_computer.rs [hunk 1]
```diff
@@ -37,6 +37,7 @@ type NotificationType = (
     Vec<ContractEvent>,
 );
 
+#[allow(dead_code)]
 type CommitType = (u64, Round, Vec<Payload>);
```
Comment (bchocho) [inline consensus/src/state_computer.rs:41]: You can actually remove this. It's getting removed in #6055

**Review by bchocho — APPROVED** [review]
Sorry about this. It'll be gone soon, but we can also land this in the interim. See below, we can get rid of CommitType

File: consensus/src/payload_manager.rs [hunk 2]
```diff
@@ -20,6 +20,7 @@ use tokio::sync::oneshot;
 /// If QuorumStore is enabled, has to ask BatchReader for the transaction behind the proofs of availability in the payload.
 pub enum PayloadManager {
     DirectMempool,
+    #[allow(dead_code)]
     InQuorumStore(BatchReader, Mutex<Sender<PayloadRequest>>),
```
Comment (gregnazario) [inline consensus/src/payload_manager.rs:24]: Let's remove rather than add dead code ignores

**Comment by banool** [comment]
I'll just land as-is and let you remove it in your PR, I don't wanna mess with anything 😛

**Pull request merged**